src/app-layer-parser.c

```diff
@@ -655,7 +662,7 @@ uint64_t AppLayerTransactionGetActiveLogOnly(Flow *f, uint8_t flags)
     AppLayerParserProtoCtx *p = &alp_ctx.ctxs[f->protomap][f->alproto];
 
     if (p->logger == TRUE) {
```

Comment:
Ok, I'll see what I can do. I don't think it can get any simpler though - only more complex if we have multiple outstanding transactions partially logged.

But can we agree on the scope here.  Being that the problem we have right now is we want to log the request and response as seen, rather than on TX completion, but within the scope of a transaction. And not the issue of logging multiple requests before the responses are seen, which I don't think we have any loggers attempting to do right now - which is going to require even more invasive changes in the case of an overlapping TX.